src/main/java/seedu/address/model/statistics/StatisticDataList.java

```diff
@@ -70,7 +69,20 @@ public ObservableList<Data> toPieChartData() {
         }
         return pieChartData;
     }
+
     public ObservableList<Data> asUnmodifiableObservableList() {
         return FXCollections.unmodifiableObservableList(this.toPieChartData());
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (other instanceof StatisticDataList) {
+            return this.getChartDataList()
+                    .equals(((StatisticDataList) other).getChartDataList());
```

Comment:
Looks like the `getChartDataList()` method doesn't exist! That's what's causing the GitHub CI to fail for this PR